pkg/forwarder/transaction/transaction.go

```diff
@@ -384,3 +385,26 @@ func (t *HTTPTransaction) SerializeTo(serializer TransactionsSerializer) error {
 	log.Trace("The transaction is not stored on disk because `storableOnDisk` is false.")
 	return nil
 }
+
+// readResponseBody read HTTP response body. If body is more than 1000 bytes
+// truncates it to  prevent from logging a huge message. When EOF occurs,
+// "<nil>" will be returned.
+// 1000 bytes are enough to tell which tools a response comes.
+func readResponseBody(body io.ReadCloser) ([]byte, error) {
+	if body == nil {
+		return []byte("<nil>"), nil
```

Comment:
Returning `<nil>` is a good idea?
It would be helpful when it clarifies that no message from a server.

Reply:
You should probably interpret the empty result in the caller otherwise the function won't really be reusable/portable in the future. If you wrap your format strings for body (`%q`s) above with single quotes it should be obvious that the body is empty I think but it's up to you.

Reply:
Return `[]byte{}` instead of `"<nil>"`. https://github.com/DataDog/datadog-agent/pull/10100/commits/ef36cfbb1298d54f9bf9df1126b46827e01fbac3